### ISSUE TITLE
support of Torch > 1.1

### ISSUE THREAD

**Issue by JerryLauzzz** [microsoft/COCO-LM#6]
[Code Ref](https://github.com/microsoft/COCO-LM/blob/92ef82ff5ac48b13b6f8bf1904919f274c180942/fairseq/fused_ops/csrc/xentropy/xentropy_kernel.cu#L79)

The use of ``THC`` is not compatible with torch>1.1.

Is there any plan to support torch>1.1?

**Comment by JerryLauzzz** [comment]
Sorry, the use of THC is not compatible with torch>1.11, instead 1.1. The problem I met is the same as [HERE](https://discuss.pytorch.org/t/question-about-thc-thc-h/147145/8).
I managed to compile the fused_ops with torch==1.11, and found that I would succeed if I made the following changes.
1. remove all include about "THC" in *.cu files.
2. change ```THCudaCheck(cudaGetLastError())``` to ```AT_CUDA_CHECK(cudaGetLastError())```.
3. change the include of ```THC/THCDeviceUtils.cuh``` to ```ATen/cuda/DeviceUtils.cuh```.
4. remove ```extern THCState *state``` in softmax_dropout_kernel.cu.

**Comment by yumeng5** [comment]
Hi,

Thanks for providing more details. I have limited knowledge about the THC part so I couldn't provide any useful information about it. I'm not sure if making those changes will somehow change the code behavior (hopefully not). The safest way probably is to create a virtual environment and run with pytorch <= 1.9.0 (I remembered this is the latest version that I have tested). Thank you anyways for providing the information for making the package compatible with newer pytorch versions.

Best,
Yu
